File: moderate/point_in_circle.rb
def in_circle(a, b, c)
  a*a + b*b <= c*c
end

File.open(ARGV[0]).each_line do |line|
  s = line.scan(/\(([-0-9.]+), ([-0-9.]+)\); Radius: ([-0-9.]+); Point: \(([-0-9.]+), ([-0-9.]+)/)[0].map(&:to_f)
  puts in_circle(s[0]-s[3], s[1]-s[4], s[2])
end

File: moderate/point_in_circle.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func inCircle(a, b, c float64) bool {
	return a*a+b*b <= c*c
}

func main() {
	data, err := os.Open(os.Args[1])
	if err != nil {
		log.Fatal(err)
	}
	defer data.Close()
	scanner := bufio.NewScanner(data)
	for scanner.Scan() {
		f := make([]float64, 5)
		fmt.Sscanf(scanner.Text(), "Center: (%f, %f); Radius: %f; Point: (%f, %f)",
			&f[0], &f[1], &f[2], &f[3], &f[4])
		if inCircle(f[0]-f[3], f[1]-f[4], f[2]) {
			fmt.Println("true")
		} else {
			fmt.Println("false")
		}
	}
}
